processor/otel/consumer.go

```diff
@@ -58,3 +74,437 @@ func (c *Consumer) ConsumeTraceData(ctx context.Context, td consumerdata.TraceDa
+				Id:        spanID,
+				ParentId:  parentID,
+				TraceId:   traceID,
+				Timestamp: startTime,
+				Duration:  duration,
+				Name:      &name,
+			}
+			parseTransaction(span, hostname, &event)
+			transformables = append(transformables, &event)
+		} else {
+			event := model_span.Event{
+				Id:        spanID,
+				ParentId:  *parentID,
+				TraceId:   traceID,
+				Timestamp: startTime,
+				Duration:  duration,
+				Name:      name,
+			}
+			parseSpan(span, &event)
+			transformables = append(transformables, &event)
+		}
+	}
+	return md, transformables
+}
+
+func parseMetadata(td consumerdata.TraceData, md *metadata.Metadata) {
+	serviceName := td.Node.GetServiceInfo().GetName()
+	if serviceName == "" {
+		serviceName = "<unknown>"
+	}
+	serviceName = truncate(serviceName)
+	md.Service = &metadata.Service{Name: &serviceName}
+
+	if ident := td.Node.GetIdentifier(); ident != nil {
+		if pid := ident.Pid; pid != 0 {
+			md.Process = &metadata.Process{Pid: int(pid)}
+		}
+		if hostname := truncate(ident.HostName); hostname != "" {
+			md.System = &metadata.System{DetectedHostname: &hostname}
+		}
+	}
+	if languageName, ok := languageName[td.Node.GetLibraryInfo().GetLanguage()]; ok {
+		md.Service.Language.Name = &languageName
+	}
+
+	switch td.SourceFormat {
+	case sourceFormatJaeger:
+		jaegerVersion := truncate(strings.TrimPrefix(td.Node.GetLibraryInfo().GetExporterVersion(), agentNameJaeger+"-"))
+		if jaegerVersion == "" {
+			jaegerVersion = "<unknown>"
+		}
+		agentName := agentNameJaeger
+		if md.Service.Language.Name != nil {
+			agentName = truncate(agentName + "/" + *md.Service.Language.Name)
+		}
+		md.Service.Agent.Name = &agentName
+		md.Service.Agent.Version = &jaegerVersion
+		if attributes := td.Node.GetAttributes(); attributes != nil {
+			if clientUUID, ok := attributes["client-uuid"]; ok {
+				clientUUID = truncate(clientUUID)
+				md.Service.Agent.EphemeralId = &clientUUID
+				delete(td.Node.Attributes, "client-uuid")
+			}
+			if ip, ok := attributes["ip"]; ok {
+				if md.System == nil {
+					md.System = &metadata.System{}
+				}
+				md.System.IP = utility.ParseIP(ip)
+				delete(td.Node.Attributes, "ip")
+			}
+		}
+	default:
+		agentName := strings.Title(td.SourceFormat)
+		md.Service.Agent.Name = &agentName
+		version := "<unknown>"
+		md.Service.Agent.Version = &version
+	}
+
+	if md.Service.Language.Name == nil {
+		unknown := "<unknown>"
+		md.Service.Language.Name = &unknown
+	}
+
+	md.Labels = make(common.MapStr)
+	for key, val := range td.Node.GetAttributes() {
+		md.Labels[key] = truncate(val)
+	}
+	if t := td.Resource.GetType(); t != "" {
+		md.Labels["resource"] = truncate(t)
+	}
+	for key, val := range td.Resource.GetLabels() {
+		md.Labels[key] = truncate(val)
+	}
+}
+
+func parseTransaction(span *tracepb.Span, hostname string, event *transaction.Event) {
+	labels := make(common.MapStr)
+
+	if count := span.GetChildSpanCount(); count != nil {
+		labels["span_count.started"] = count.Value
```

Comment:
I'm not sure if this is a good idea, as I believe the semantics are different. IIANM, `span_count.started` is meant to record how many (child, grandchild, etc.) spans were started within this transaction, whereas Jaeger/OpenTelemetry just tells us the number of direct child spans started.

_Maybe_ we could redefine `span_count.started` to be a lower bound on the number of started spans.

Reply:
I only added it as labels, but can remove overall to avoid confusion.

Reply:
Oops, I didn't read that properly. I think removing it's the right way to go.